lib/dependencies/SystemPlugin.js

```diff
@@ -39,11 +43,35 @@ class SystemPlugin {
 						parser.state.module.context, require.resolve("../../buildin/system.js"));
 					return ParserHelpers.addParsedVariableToModule(parser, "System", systemPolyfillRequire);
 				});
+
+				parser.hooks.call.for("System.import").tap("SystemPlugin", expr => {
+					if(shouldWarn) {
+						shouldWarn = false;
+						compilation.warnings.push(new SystemImportDeprecationWarning());
+					}
+
+					parser.hooks.importCall.call(expr);
+					return true;
+				});
 			};
 
 			normalModuleFactory.hooks.parser.for("javascript/auto").tap("SystemPlugin", handler);
 			normalModuleFactory.hooks.parser.for("javascript/dynamic").tap("SystemPlugin", handler);
 		});
 	}
 }
+
+class SystemImportDeprecationWarning extends WebpackError {
+	constructor(modules) {
+		super();
+
+		this.name = "SystemImportDeprecationWarning";
+		this.message = "System.import() is deprecated and will be removed soon.\n" +
+			"Use import() or require.ensure() instead.\n" +
+			"For more info visit https://webpack.js.org/guides/code-splitting/";
```

Comment:
This should print a list of all modules using System.import.

Reply:
What about adding this to the warning and allow multiple emit?

```js
this.origin = this.module = module;
this.originLoc = loc;
```

The above modification emits the following warnings:

```
System.import() is deprecated and will be removed soon. Use import() instead.
For more info visit https://webpack.js.org/guides/code-splitting/
 @ ./index.js 43:2-33
```

It may be better to debug these messages rather than a module list. What do you think?

Reply:
Should I remove `will be removed soon`? I feel like it's a bit to much.

Reply:
yep, that's better.

Add the warning to `parser.scope.module.warnings` instead of to the Compilation directly. This should ensure that it's correctly cached.

Reply:
`Stats` does not extract `originLoc` of `Module#warnings`; only `Compilation#warnings`. I'm unable to match the _location_ into `warnings.js` when using `parser.scope.module` instead of `compilation`.